pkg/ipcache/ipcache.go

```diff
@@ -286,7 +287,7 @@ func ipIdentityWatcher(listeners []IPIdentityMappingListener) {
 
 				// Callback upon cache updates.
 				for _, listener := range listeners {
-					if cachedIdentity != 0 && ipIDPair.ID != cachedIdentity {
+					if ipIsInCache && cacheModification == Upsert {
```

Comment:
Shouldn't this be `cacheModification == Delete`?

Reply:
No, this is to delete the old value in case of an update (not insert).
The delete case is handled by the next call below.

Reply:
I'll add a comment.

Reply:
change the `cacheModification` name to `cacheOperation`, I think it will help

Reply:
Done.